api/python/quilt3/util.py

```diff
@@ -433,3 +433,11 @@ def validate_key(key):
                 f"Invalid key {key!r}. "
                 f"A package entry key cannot contain a file or folder named '.' or '..' in its path."
             )
+
+def catalog_s3_url(s3_url):
```

Comment:
Just a thought - but might be cleaner to pass in the catalog URL, and have this function return the full URL, not just the path and query params.